src/OpenTelemetry.Collector.AspNetCore/AssemblyInfo.cs

```diff
@@ -15,4 +15,8 @@
 // </copyright>
 using System.Runtime.CompilerServices;
 
+#if SIGNED
 [assembly: InternalsVisibleTo("OpenTelemetry.Collector.AspNetCore.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010051c1562a090fb0c9f391012a32198b5e5d9a60e9b80fa2d7b434c9e5ccb7259bd606e66f9660676afc6692b8cdc6793d190904551d2103b7b22fa636dcbb8208839785ba402ea08fc00c8f1500ccef28bbf599aa64ffb1e1d5dc1bf3420a3777badfe697856e9d52070a50c3ea5821c80bef17ca3acffa28f89dd413f096f898")]
+#else
+[assembly: InternalsVisibleTo("OpenTelemetry.Collector.AspNetCore.Tests")]

```

Comment:
I saw that the Linux build was failing because the assemblies are not signed at all except when the build agent is Windows. In lieu of signing under those circumstances I wrapped the InternalsVisibleTo attribute in an appropriate preprocessor directive.